File: projects/engineering.purdue.edu/programmingwithobjects/pwocode/chap5/SetOps.java
// This code example is from the following source:
//
// Book Title:  Programming with Objects, A Comparative Presentation
//              of Object-Oriented Programming with C++ and Java
//
// Book Author:  Avinash Kak
//
// Chapter:     Chapter 5 ---- Using the Container Classes
//
// Section:     Section 5.2.2 ---- Set
//





//SetOps.java

import java.util.*;

class SetOps {

    public static void main( String[] args )
    {
        Set animals = new TreeSet();                              //(A)

        animals.add( "cheetah" );                                 //(B)
        animals.add( "lion" );                                    //(C)
        animals.add( "cat" );                                     //(D)
        animals.add( "elephant" );                                //(E)
        animals.add( "cat" );                  // duplicate cat   //(F)
        System.out.println( animals );                            //(G)
                                // cat cheetah elephant lion
        System.out.println( animals.size() );  // 4               //(H)

        animals.remove( "lion" );                                 //(I}
        System.out.println( animals );  // cat cheetah elephant   //(J)

        Iterator iter = animals.iterator();                       //(K)
        while ( iter.hasNext() )                                  //(L)
            System.out.println( iter.next() );                    //(M)
                                        // cat cheetah elephant
    }
}

File: projects/demos/engineering.purdue.edu/programmingwithobjects/pwocode/chap5/SetOps.cc
// This code example is from the following source:
//
// Book Title:  Programming with Objects, A Comparative Presentation
//              of Object-Oriented Programming with C++ and Java
//
// Book Author:  Avinash Kak
//
// Chapter:     Chapter 5 ---- Using the Container Classes
//
// Section:     Section 5.1.8  ---- Set
//


//SetOps.cc

#include <iostream>                      // for cout, endl
#include <string>
#include <set>                                                    //(A)
using namespace std;

int main()
{
    set<string> animals;                                          //(B)

    animals.insert( "cheetah" );                                  //(C)
    animals.insert( "lion" );                                     //(D)
    animals.insert( "cat" );                                      //(E)
    animals.insert( "elephant" );                                 //(F)
    animals.insert( "cat" );             //attempting a duplicate //(G)

    cout <<  animals.size() << endl;;    // 4                     //(H)
                                                             
    typedef set<string>::const_iterator CI;                  
    for (CI iter = animals.begin();                               //(I)
            iter != animals.end(); 
            iter++) 
        cout << *iter << " ";            // cat cheetah elephant lion

    animals.erase( "lion" );                                      //(J)
    cout <<  animals.size() << endl;;    // 3                     //(K)

    for ( CI iter = animals.begin();                              //(L)
          iter != animals.end(); 
          iter++ )
        cout << *iter << " ";            // cat cheetah elephant

    return 0;
}
